### PR TITLE
Add backwards compatibility with v0.3.x

### PR DESCRIPTION
**Pull request by llucax**
This PR includes the top-level files present in the v0.3.x release, so the Microgrid API v0.15.x can depend on a newer version of the Common API.

When the Microgrid API v0.15.x is phased out we can remove all the files related to v0.3.x from this repository.

### CODE REVIEW

File: proto/frequenz/api/common/components.proto [hunk 1]
```diff
@@ -0,0 +1,132 @@
+// Frequenz microgrid components definitions.
+//
+// Copyright:
+// Copyright 2023 Frequenz Energy-as-a-Service GmbH
```
Comment (tiyash-basu-frequenz) [inline proto/frequenz/api/common/components.proto:4]: Should we put 2024 as the year?

Comment (llucax) [reply proto/frequenz/api/common/components.proto:4]: I don't think so, this file was created in 2023 originally.

**Comment by llucax** [comment]
Enabled auto-merge.

**Review by TalweSingh — APPROVED** [review]

**Comment by llucax** [comment]
@tiyash-basu-frequenz we proceeded because you gave your blessing before, you only posted one small comment, and we want to fix this dependency issue ASAP, but if you have any further objections we can address them in a separate PR/release.

**Pull request merged**